kornia/contrib/sam/model.py

```diff
@@ -0,0 +1,196 @@
+        """Drop the passed index for all data.
+
+        Performs `self.prop = self.prop[idx]` for each property
+        """
+        self.masks = self.masks[idx]
+        self.scores = self.scores[idx]
+        self.logits = self.logits[idx]
+
+        return self
+
+
+def load_sam_vit_h(checkpoint: str | None = None, device: torch.device | None = None) -> Sam:
+    return _build_sam(
+        encoder_embed_dim=1280,
+        encoder_depth=32,
+        encoder_num_heads=16,
+        encoder_global_attn_indexes=(7, 15, 23, 31),
+        checkpoint=checkpoint,
+        device=device,
+    )
+
+
+def load_sam_vit_l(checkpoint: str | None = None, device: torch.device | None = None) -> Sam:
+    return _build_sam(
+        encoder_embed_dim=1024,
+        encoder_depth=24,
+        encoder_num_heads=16,
+        encoder_global_attn_indexes=(5, 11, 17, 23),
+        checkpoint=checkpoint,
+        device=device,
+    )
+
+
+def load_sam_vit_b(checkpoint: str | None = None, device: torch.device | None = None) -> Sam:
+    return _build_sam(
+        encoder_embed_dim=768,
+        encoder_depth=12,
+        encoder_num_heads=12,
+        encoder_global_attn_indexes=(2, 5, 8, 11),
+        checkpoint=checkpoint,
+        device=device,
+    )
+
+
+class LayerNorm(nn.LayerNorm):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.eps = 1e-6
+
+
+def _build_sam(
+    encoder_embed_dim: int,
+    encoder_depth: int,
+    encoder_num_heads: int,
+    encoder_global_attn_indexes: tuple[int, ...],
+    checkpoint: str | None = None,
+    device: torch.device | None = None,
+) -> Sam:
+    prompt_embed_dim = 256
+    image_size = 1024
+    vit_patch_size = 16
+    image_embedding_size = image_size // vit_patch_size
+
+    sam = Sam(
+        image_encoder=ImageEncoderViT(
+            depth=encoder_depth,
+            embed_dim=encoder_embed_dim,
+            img_size=image_size,
+            mlp_ratio=4,
+            norm_layer=LayerNorm,
+            num_heads=encoder_num_heads,
+            patch_size=vit_patch_size,
+            qkv_bias=True,
+            use_rel_pos=True,
+            global_attn_indexes=encoder_global_attn_indexes,
+            window_size=14,
+            out_chans=prompt_embed_dim,
+        ),
+        prompt_encoder=PromptEncoder(
+            embed_dim=prompt_embed_dim,
+            image_embedding_size=(image_embedding_size, image_embedding_size),
+            input_image_size=(image_size, image_size),
+            mask_in_chans=16,
+        ),
+        mask_decoder=MaskDecoder(
+            num_multimask_outputs=3,
+            transformer=TwoWayTransformer(depth=2, embedding_dim=prompt_embed_dim, mlp_dim=2048, num_heads=8),
+            transformer_dim=prompt_embed_dim,
+            iou_head_depth=3,
+            iou_head_hidden_dim=256,
+        ),
+        pixel_mean=[123.675, 116.28, 103.53],
+        pixel_std=[58.395, 57.12, 57.375],
+        device=device,
```

Comment:
Is really device needed since it’s a Module and the .to semantics affects to all the internal modules ?

Reply:
Nops, i added it to test something, but didn't worked, needs to remove it